File: output/ChatGPT/avatar/Java/C++/codeforces_665_A.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    int a, tA, b, tB;
    cin >> a >> tA >> b >> tB;

    string timing;
    cin >> timing;
    int hrs = stoi(timing.substr(0, 2));
    int mins = stoi(timing.substr(3));
    int simDeparture = hrs * 60 + mins;
    int simArrival = simDeparture + tA;
    int counter = 0;

    for (int i = 300; i < 1440; i += b) {
        int busDeparture = i;
        int busArrival = i + tB;
        if (busDeparture >= simArrival || simDeparture >= busArrival)
            continue;
        counter++;
    }

    cout << counter << endl;

    return 0;
}

File: output/ChatGPT/avatar/Java/Python/codeforces_665_A.py
import sys

def main():
    a, tA = map(int, sys.stdin.readline().split())
    b, tB = map(int, sys.stdin.readline().split())
    timing = sys.stdin.readline().strip()
    hrs = int(timing[:2])
    mins = int(timing[3:])
    simDeparture = hrs * 60 + mins
    simArrival = simDeparture + tA
    counter = 0
    for i in range(300, 1440, b):
        busDeparture = i
        busArrival = i + tB
        if busDeparture >= simArrival or simDeparture >= busArrival:
            continue
        counter += 1
    print(counter)

if __name__ == "__main__":
    main()